quic/connection.nim

```diff
@@ -152,6 +154,28 @@ proc newOutgoingConnection*(
 proc startHandshake*(connection: Connection) {.gcsafe.} =
   connection.quic.send()
 
+proc waitForHandshake*(
+    connection: Connection
+) {.async: (raises: [CancelledError, TimeOutError, CatchableError]).} =
+  let key = connection.quic.error.register()
+  defer:
+    connection.quic.error.unregister(key)
+
+  let errFut = connection.quic.error.waitEvents(key)
+  let timeoutFut = connection.quic.timeout.wait()
+  let handshakeFut = connection.quic.handshake.wait()
+  let race = await race(handshakeFut, timeoutFut, errFut)
+  if race == timeoutFut:
+    await allFutures(timeoutFut, connection.close())
+    errFut.cancelSoon()
+    raise newException(TimeOutError, "handshake timed out")
+  elif race == errFut:
+    let err = await errFut
+    if err[0] != "":
```

Comment:
would be nice to explain why `errFut` completes with empty error.

Reply:
nice catch. This should no longer happen, as `errFut` will always contain an error message now, or not be emitted.